core/templates/expressions/expression-syntax-tree.service.ts

```diff
@@ -16,417 +16,488 @@
+        throw new ExprWrongArgTypeError(null, argTypes[i], expectedType);
       }
-      return true;
-    };
+    }
+    return true;
+  }
 
-    var applyFunctionToParseTree = function(parsed, envs, func) {
-      return func(parsed, envs.concat(system));
-    };
+  private _verifyArgTypesMatch(argType1: string, argType2: string): boolean {
+    if (argType1 !== argType2) {
+      throw new ExprWrongArgTypeError(null, argType1, argType2);
+    }
+    return true;
+  }
 
-    /**
-     * Looks up a variable of the given name in the env. Here the variable
-     * can be system or user defined functions and parameters, as well as
-     * system operators.
-     * @param {string} name The name to look up.
-     * @param {!Array.<!object>} envs Represents a nested name space
-     *     environment to look up the name in. The first element is looked up
-     *     first (i.e. has higher precedence).
-     * @throws {ExprUndefinedVarError} The named variable was not found in
-     *     the given environment.
-     */
-    const lookupEnvs = (name: string, envs: object[]) => {
-      for (const env of envs) {
-        if (env.hasOwnProperty(name)) {
-          return env[name];
-        }
+  public applyFunctionToParseTree(
+      parsed: string,
+      envs: Array<object>,
+      func: (parsed: string, envs: Array<any>) => string
+  ) {
+    return func(parsed, envs.concat(this.system));
+  }
+
+  /**
+   * Looks up a variable of the given name in the env. Here the variable
+   * can be system or user defined functions and parameters, as well as
+   * system operators.
+   * @param {string} name The name to look up.
+   * @param {!Array.<!object>} envs Represents a nested name space
+   *     environment to look up the name in. The first element is looked up
+   *     first (i.e. has higher precedence).
+   * @throws {ExprUndefinedVarError} The named variable was not found in
+   *     the given environment.
+   */
+  public lookupEnvs(name: string, envs: Array<object>): string {
+    for (const env of envs) {
+      if (env.hasOwnProperty(name)) {
+        return env[name];
       }
+    }
 
-      throw new ExprUndefinedVarError(name, envs);
-    };
+    throw new ExprUndefinedVarError(name, envs);
+  }
 
-    // Coerces the argument to a Number, and throws an error if the result
-    // is NaN.
-    var _coerceToNumber = function(originalValue) {
-      var coercedValue = (+originalValue);
-      if (!isNaN(coercedValue)) {
-        return coercedValue;
-      }
-      throw new ExprWrongArgTypeError(
-        originalValue, typeof originalValue, 'Number');
-    };
+  // Coerces the argument to a Number, and throws an error if the result
+  // is NaN.
+  private _coerceToNumber(originalValue: string): number {
+    var coercedValue = +originalValue;
+    if (!isNaN(coercedValue)) {
+      return coercedValue;
+    }
+    throw new ExprWrongArgTypeError(
+      originalValue,
+      typeof originalValue,
+      'Number'
```

Comment:
Can we move some of the args in the same line here as well?

Reply:
Ditto below, I see each param in new line and then closing bracket in next line. Can we try to move as many params in one line and the closing bracket as well in the same ending line?

Reply:
@bansalnitish PTAL!